script.js

```diff
@@ -1,4 +1,348 @@
+//
+// MAKE DECK
+//
+//
+var makeDeck = function () {
+  var cardDeck = [];
+
+  // Initialise an array of the 4 suits in our deck. We will loop over this array.
+  var suits = ["hearts", "diamonds", "clubs", "spades"];
+
+  // Loop over the suits array
+  var suitIndex = 0;
+  while (suitIndex < suits.length) {
+    // Store the current suit in a variable
+    var currentSuit = suits[suitIndex];
+
+    // Loop from 1 to 13 to create all cards for a given suit
+    // Notice rankCounter starts at 1 and not 0, and ends at 13 and not 12.
+    // This is an example of a loop without an array.
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      // By default, the card name is the same as rankCounter
+      var cardName = rankCounter;
+      var cardValue = rankCounter;
+
+      if (cardName == 1) {
+        cardName = "ace";
+        cardValue = 1;
+      } else if (cardName == 11) {
+        cardName = "jack";
+        cardValue = 10;
+      } else if (cardName == 12) {
+        cardName = "queen";
+        cardValue = 10;
+      } else if (cardName == 13) {
+        cardName = "king";
+        cardValue = 10;
+      }
+
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        score: cardValue,
+      };
+
+      cardDeck.push(card);
+      rankCounter += 1;
+    }
+    suitIndex += 1;
+  }
+  return cardDeck;
+};
+//
+//
+//
+//
+//
+//
+//
+// SHUFFLE DECK
+//
+//
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// Shuffle the elements in the cardDeck array
+var shuffleCards = function (cardDeck) {
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // Increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // Return the shuffled deck
+  return cardDeck;
+};
+//
+//
+//
+//
+//
+//
+//
+//START MAIN FUNCTION
+//
+//
+var shuffledDeck = shuffleCards(makeDeck());
+var playerDeck = shuffledDeck.splice(0, 26);
+var computerDeck = shuffledDeck;
```

Comment:
Ah, so you are taking the first 26 cards and assigning it to the player. Then, taking the rest of the other 26 cards and giving it to the computer?

You can stick to the current arrangement for the presentation later, but in the game of Blackjack, it might be better for the player and computer to draw from the same deck.

This is because even if the computer and player decks are evenly split, it is uncertain that both decks will deplete at the same rate.

In your interface, the player and computer can end up drawing a different number of cards per round: a computer will continue hitting until it is close to 21 or beyond, and so will the player. This means the hands of the computer and the player may have different number of cards by the time it's time to evaluate their final scores.

By splitting evenly, we know that one of the decks will deplete faster, and the game will end a lot quicker. Let's assume that either the computer or the player is really unlucky throughout the rounds, one of them will end up depleting his deck really quickly, while the other still has plenty of cards left in their deck.